src/video/videosurface.cpp

```diff
@@ -68,9 +70,14 @@ void VideoSurface::setSource(VideoSource *src)
     if (source == src)
         return;
 
-    unsubscribe();
-    source = src;
-    subscribe();
+    // To don't block GUI thread.
+    QtConcurrent::run([this, src]()

```

Comment:
`bigLock`, `(un)subscribe`, `source` non-static.
They can't be called without object.